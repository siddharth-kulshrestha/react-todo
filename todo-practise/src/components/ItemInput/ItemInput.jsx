import React, { PureComponent } from 'react';
import './iteminput.css';

export default class ItemInput extends PureComponent {
    state = {
        content: "",
        rating: 0,
    }

    handleContentChange(event) {
        this.setState({content: event.target.value});
    }

    handleRatingChange(event) {
        this.setState({rating: event.target.value});
    }

    handleOnSubmit(event) {
        event.preventDefault();
        this.props.onInput({content: this.state.content, rating: this.state.rating});
    }

    render() {
        return (
            <form className="form-container" onSubmit={this.handleOnSubmit.bind(this)}>
                <input 
                type="text"
                placeholder="Item Content"
                value={this.state.content}
                onChange={this.handleContentChange.bind(this)}
                />
                <input 
                type="number"
                placeholder="Rating"
                value={this.state.rating}
                onChange={this.handleRatingChange.bind(this)}
                />
                <input
                type="submit"
                value="Add"
                />
            </form>
        );
    }
}
