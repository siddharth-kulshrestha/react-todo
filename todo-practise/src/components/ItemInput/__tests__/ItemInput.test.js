import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ItemInput from '../ItemInput';

configure({adapter: new Adapter()});

const mockItem = {
    id: 0,
    item: {
        content: "test content",
        rating: 6,
    }
};

// Snapshot test for ItemInput component
describe('Item Input ', () => {
    it('should render correctly ', () => {
        const component = shallow(<ItemInput onInput={(item) => (item)} />);
        expect(component).toMatchSnapshot();
    });

    it('should simulate and render item input correctly ', () => {
        const onInputMock = jest.fn();
        const component = shallow(<ItemInput onInput={onInputMock} />);
        component.setState({ content: "Test content", rating: 10 });
        const content = component.findWhere((comp) => comp.type() == 'input' && comp.props().type === 'text').props().value;
        const rating = component.findWhere((comp) => comp.type() == 'input' && comp.props().type === 'number').props().value;
        expect(content).toEqual('Test content');
        expect(rating).toEqual(10);

        const form = component.find('form')
        form.simulate('submit', {
            preventDefault: () => {
            }
        }); // second argument is empty event object which will get sent when form is submit in running simulation.
        expect(onInputMock).toBeCalledWith({content: "Test content",  rating: 10});
        component.unmount();
    });
});
