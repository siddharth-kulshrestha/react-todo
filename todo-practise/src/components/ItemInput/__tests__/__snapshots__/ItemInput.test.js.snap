// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Input  should render correctly  1`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Item Content"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Rating"
    type="number"
    value={0}
  />
  <input
    type="submit"
    value="Add"
  />
</form>
`;
